src/Microsoft.Extensions.PlatformAbstractions/Native/NativeMethods.Unix.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.PlatformAbstractions.Native
+{
+    internal static partial class NativeMethods
+    {
+        public static class Unix
+        {
+            public unsafe static string GetUname()
+            {
+                // Utsname shouldn't be larger than 2K
+                var buf = stackalloc byte[2048];
+
+                try
+                {
+                    if (uname((IntPtr)buf) != 0)
+                    {
+                        throw new PlatformNotSupportedException("Error reading Unix Name");
```

Comment:
Maybe. `uname` is short for "Unix Name" as a proper noun... but yeah, I'll probably just lcase it.